src/doc/rustc/src/instrument-coverage.md

```diff
@@ -97,7 +97,24 @@ $ echo "{some: 'thing'}" | target/debug/examples/formatjson5 -
 }
 ```
 
-After running this program, a new file, `default.profraw`, should be in the current working directory. It's often preferable to set a specific file name or path. You can change the output file using the environment variable `LLVM_PROFILE_FILE`:
+After running this program, a new file, `default_%m_%p.profraw`, should be in the current working directory. This file takes advantage ofLLVM's support for rewriting special pattern strings to ensure `.profraw` files generated are unique. The following special pattern strings are rewritten as:
+
+-   `%p` - The process ID.
+-   `%h` - The hostname of the machine running the program.
+-   `%t` - The value of the TMPDIR environment variable.
+-   `%Nm` - the instrumented binary’s signature: The runtime creates a pool of N raw profiles, used for on-line profile merging. The runtime takes care of selecting a raw profile from the pool, locking it, and updating it before the program exits. `N` must be between `1` and `9`, and defaults to `1` if omitted (with simply `%m`).
+-   `%c` - Does not add anything to the filename, but enables a mode (on some platforms, including Darwin) in which profile counter updates are continuously synced to a file. This means that if the instrumented program crashes, or is killed by a signal, perfect coverage information can still be recovered.
```

Comment:
This is pretty good but it kind of reads to me like the file name is (exactly) `default_%m_%p.profraw`. I would suggest two things:

1. Change the first sentence to say something more general like "a new file named like `default_1234_5678.profraw` should be in the current working directory. A new, unique file name will be generated each time the program is run to avoid overwriting previous data." (or actually, use the file name in the sentence that you're using in the example so they match)
2. Move the explanation about the formatting options back down to the mention of `LLVM_PROFILE_FILE`.

Reply:
Thanks for the feedback. I've updated the documentation to make it clearer on the file naming scheme moving forward.